### PR TITLE
types: import types for `@nuxt/telemetry` and `@nuxt/components`

### PR DESCRIPTION
**Pull request by danielroe**
Add missing types for new Nuxt modules `@nuxt/telemetry` and `@nuxt/components`.

## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Added two new properties to `NuxtConfig` type.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Review by pi0 — DISMISSED** [review]

File: packages/types/config/index.d.ts [hunk 1]
```diff
@@ -32,6 +34,7 @@ export interface NuxtOptions extends Configuration {
   buildDir: string
   buildModules: NuxtOptionsModule[]
   cli: NuxtOptionsCli
+  components: boolean | Array<string | ComponentsDir>
```
Comment (pi0) [inline packages/types/config/index.d.ts:37]: There is also object format support (which may be used later on to provide more options

Comment (danielroe) [reply packages/types/config/index.d.ts:37]: Ideally the options would be completely defined in `@nuxt/components` and just imported in `@nuxt/types`. I will propose a PR to do that if that sounds good to you.

Comment (pi0) [reply packages/types/config/index.d.ts:37]: Indeed. Just published types for both :see_no_evil: Updating deps in a sec.

Comment (pi0) [reply packages/types/config/index.d.ts:37]: #7621

**Pull request merged**